aten/src/ATen/native/cuda/MiscUtils.h

```diff
@@ -0,0 +1,50 @@
+#include "ATen/ATen.h"
+#include "THC.h"  // for USE_MAGMA
+
+#ifdef USE_MAGMA
+#include <magma.h>
+#include <magma_types.h>
+#endif
+
+namespace at {
+namespace native {
+
+#ifdef USE_MAGMA
+static magma_queue_t createMagmaQueue(const Tensor& tensor) {
```

Comment:
We should make this RAII given that we always want to delete the created queue

Reply:
We can make this RAII if there is no considerable overhead associated with creating a queue (RAII will mandate the creation of two queues, one for `getri` and one for `getrf`).

Reply:
Sorry, I'm not following: the magma queue seems local to the function, so wouldn't getri and getrf use different queues regardless of if magma queue creation is RAII or not?

To clarify with gesv as an example, the code currently looks like this:
```
  magma_queue_t gesv_magma_queue = createMagmaQueue(b);
   magmaGesvBatched<scalar_t>(
      n, nrhs, A_array, n, ipiv_array, b_array, n,
      info_array, batch_size, gesv_magma_queue);
   destroyMagmaQueue(gesv_magma_queue);
```
but we could prevent future contributors from accidentally forgetting to free the queue with something like the following:
```
  MagmaQueue gesv_magma_queue;
   magmaGesvBatched<scalar_t>(
      n, nrhs, A_array, n, ipiv_array, b_array, n,
      info_array, batch_size, gesv_magma_queue);
  // no need to destory gesv_magma_queue, it will be cleaned up when it goes out of scope
```

Reply:
I seemed to have misunderstood your pitch earlier, and I think understand now. Thank you for clarifying.

In my earlier comment I thought of something like this:
```
   magmaGesvBatched<scalar_t>(
      n, nrhs, A_array, n, ipiv_array, b_array, n,
      info_array, batch_size, createMagmaQueue(b)); 
// This would be cleaned up, right? - since it is local to the function.
```

Reply:
Oh I see. Yes, in that case, The magma queue would only live inside the call to `magmaGesvBatched`

Reply:
I will change the functions to do this instead, and remove `destroyMagmaQueue`. Thanks for the suggestion.